Custom-ESLint-checks.md

```diff
@@ -0,0 +1,329 @@
+
+#### Run the linter with the new rule
+
+To run the linter with your new rule, follow these steps:
+
+1. Delete the `node_modules/eslint-plugin-oppia/` directory so that in the next steps, the lint rules get re-installed.
+
+2. Run `yarn install` to install the custom rules, including the one you just wrote.
+
+3. Execute the pre-commit linter, which will use your new rule. To speed up the lint checks, we recommend running the rule on just a single file when testing it. If you created a `test.js` file with some code to try linting, you would run the linter like this:
+
+   ```console
+   $ python -m scripts.linters.pre_commit_linter --path test.js
+   ...
+   ----------------------------------------
+   Please fix the errors below:
+   ----------------------------------------
+   ...
+   /oppia/test.js
+     2:11    Do not use "Hello, world!" strings  oppia/hello-world
+
+   ✖ 1 problem (1 error, 0 warnings)
+
+
+
+   ---------------------------
+   Checks Not Passed.
+   ---------------------------
+   ```
+
+   Note that you might see other lint errors too. Those are fine to ignore.
+
+### Beyond hello world
+
+ESLint has many more features described in the [ESLint documentation](https://eslint.org/docs/developer-guide/working-with-rules). In particular, you may find the following features useful:
+
+#### Placeholders
+
+Your error messages can have [placeholders](https://eslint.org/docs/developer-guide/working-with-rules#using-message-placeholders) that are filled by the data you pass to `context.report()`. For example, you might have an error message like this:
+
+```js
+literals: 'Please do not use "{{stringValue}}" literals.'
+```
+
+Then you could call `context.report()` like this:
+
+```js
+context.report({
+  node: node,
+  messageId: 'literals',
+  data: {
+    stringValue: node.value,
+  },
+})
+```
+
+The value of `node.value` will be substituted in place of `{{stringValue}}`.
+
+#### Navigating the AST
+
+You'll often want to move up or down the AST to check neighboring nodes. For example, suppose you want to forbid calling `console.log()`. `console` and `log` are different nodes, and you need to check both. Here's an abbreviated version of the AST for `console.log()`:
+
+```json
+{
+  "callee": {
+    "type": "MemberExpression",
+    ...
+    "object": {
+      "type": "Identifier",
+      ...
+      "name": "console",
+      ...
+    },
+    "property": {
+      "type": "Identifier",
+      ...
+      "name": "log",
+      ...
+    },
+    ...
+  },
+}
+```
+
+You could identify this pattern by specifying a function to run on all nodes of type `MemberExpression`. Inside your function, you could then check that `node.object.name === "console"` and `node.property.name === "log"`.
+
+It is also possible to work up the AST by calling `context.getAncestors()`. This call will return an array of all the ancestors of the current node.
+
+## Write tests
```

Comment:
Perhaps this should come before the "optional" section? Or you can say something in the intro like: in order to write a new rule, you'll need to write the rule and you'll need to test it. (Or something like "Step 1" and "Step 2" in the headers -- something to show that this section isn't just an ignorable appendix because it comes below an optional section.)

Reply:
done